src/Amazon.AspNetCore.DataProtection.SSM/SSMXmlRepository.cs

```diff
@@ -192,6 +200,96 @@ private async Task StoreElementAsync(XElement element, string friendlyName)
             }
         }
 
+#if NET9_0_OR_GREATER
+        /// <inheritdoc/>
+        public bool DeleteElements(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+        {
+            return Task.Run(() => DeleteElementsAsync(chooseElements)).GetAwaiter().GetResult();
+        }
+
+        private async Task<bool> DeleteElementsAsync(Action<IReadOnlyCollection<IDeletableElement>> chooseElements)
+        {
+            if (chooseElements == null)
+            {
+                throw new ArgumentNullException(nameof(chooseElements));
+            }
+
+            var deletableElements = new List<DeletableElement>();
+            var request = new GetParametersByPathRequest
+            {
+                Path = _parameterNamePrefix,
+                WithDecryption = true
```

Comment:
is there any use case for this to be false?

Reply:
I used the existing implementation from `GetAllElementsAsync`. In case the keys are encrypted, then it needs to decrypt them. In `StoreElementAsync`, we put parameter as `ParameterType.SecureString`.